torchao/quantization/quant_api.py

```diff
@@ -401,10 +404,25 @@ def insert_subclass(lin):
     return insert_subclass
 
 def quantize_(
+    # NOT FOR LAND: this is the same across inference and torchao.float8
     model: torch.nn.Module,
+
+    # NOT FOR LAND: this seems to leak implementation details, in the future
+    # can this be named in a more generic way without mentioning the 
+    # implementation? Short term, seems ok to have float8 training use this
+    # evern though the implementation is different from 100% tensor subclass 
+    # based workflows.
     apply_tensor_subclass: Callable[[torch.nn.Module], torch.nn.Module],
+
+    # NOT FOR LAND: this is the same across inference and torchao.float8
     filter_fn: Optional[Callable[[torch.nn.Module, str], bool]] = None,
+
+    # NOT FOR LAND: this seems inference specific, in the future would we 
+    # consider renaming or some other way to make things more clear?
     set_inductor_config: bool = True,
```

Comment:
we just used to get so many pings where it was like "its not faster" and unsurprisingly the individual didn't set the inductor config. I'm not sure the best way to handle this. Without it our apis also get 100% more complicated since its not a 1 line api anymore. We can move it into the constructors if necessary

Reply:
I'm not saying it's bad, I'm just saying if training is going to be in here then this setting could be renamed `set_inductor_config_to_recommended_inference_settings`, etc

Reply:
yeah but its still either A) not on by default or B) users of non inference use cases have to disable it always.

the more i think about it, the more it seems that having different defaults for different use cases makes sense.

Reply:
maybe just move this to apply_tensor_subclass?

Reply:
note: `set_inductor_config` is changing inductor configuration, that does not seem related to modifying modules of a model, so moving this to `apply_tensor_subclass` would be confusing to me